content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md

```diff
@@ -0,0 +1,367 @@
+> * [Kubernetes Response Engine, Part 4 : Falcosidekick + Tekton]({{< ref "/blog/falcosidekick-reponse-engine-part-4-tekton" >}})
+> * [Kubernetes Response Engine, Part 5 : Falcosidekick + Argo]({{< ref "/blog/falcosidekick-reponse-engine-part-5-argo" >}})
+> * [Kubernetes Response Engine, Part 6 : Falcosidekick + Cloud Run]({{< ref "/blog/falcosidekick-reponse-engine-part-6-cloud-run" >}})
+---
+
+Earlier in this series we have seen how to use Argo, Tekton and Knative to trigger a resource after getting input from Falcosidekick. 
+Recently, Falcosidekick got a new output type support for [Cloud Functions](https://github.com/falcosecurity/falcosidekick/pull/241). 
+
+In this part, we will learn how can use Falcosidekick + Cloud Functions to detect and delete a compromised pod.
+
+We will not go through what Cloud Functions is in depth but, you can always find a good overview in the [official documentation](https://cloud.google.com/functions).
+
+Here is the high-level overview architecture that shows what we want to achieve at the end of the day:
+
+![cloud_functions_reference_arch](/docs/images/cloud_functions_reference_arch.png)
+
+You can find all the related code and resources in [this repository](https://github.com/Dentrax/kubernetes-response-engine-based-on-gke-and-gcloudfunctions).
+
+## Prerequisites
+
+- gcloud 342.0.0
+- kubectl 1.20.5
+
+## Tutorial
+
+### Provision Google Kubernetes Engine (GKE) Cluster
+
+As the blog title said already, we need to create a [GKE cluster](https://cloud.google.com/kubernetes-engine) with workload identity enabled:
+
+```shell
+$ GOOGLE_PROJECT_ID=$(gcloud config get-value project)
+$ CLUSTER_NAME=falco-falcosidekick-demo
+$ gcloud container clusters create $CLUSTER_NAME --workload-pool ${GOOGLE_PROJECT_ID}.svc.id.goog
+$ gcloud container clusters get-credentials $CLUSTER_NAME
+```
+
+### Configuring IAM Service Accounts
+
+We need to create a new [Service Account](https://cloud.google.com/iam/docs/service-accounts) for target `$GOOGLE_PROJECT_ID` using IAM Binding [policies](https://cloud.google.com/iam/docs/policies) to get access our Cloud Function:
+```shell
+$ SA_ACCOUNT=falco-falcosidekick-sa
+$ gcloud iam service-accounts create $SA_ACCOUNT
+
+$ gcloud projects add-iam-policy-binding ${GOOGLE_PROJECT_ID} \
+--member="serviceAccount:${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com" \
+--role="roles/cloudfunctions.developer"
+
+$ gcloud projects add-iam-policy-binding ${GOOGLE_PROJECT_ID} \
+--member="serviceAccount:${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com" \
+--role="roles/cloudfunctions.invoker"
+```
+
+At the beginning, we already enabled [WorkloadIdentity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) feature for our GKE Cluster by setting `--workload-pool` flag. What we need to do here is we should add a `iam.workloadIdentityUser` role for the given Service Account. 
```

Comment:
In the beginning, we already enabled [WorkloadIdentity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) feature for our GKE Cluster by setting `--workload-pool` flag. What we need to do here is to add a `iam.workloadIdentityUser` role for the given Service Account.